### ISSUE TITLE
raygun: Stop owner-blocked users from being added to open groups

### ISSUE THREAD

**Issue by zeenix** [Satellite-im/Warp#418]
Like #417, this is also a follow-up of https://github.com/Satellite-im/Warp/issues/386. See description of https://github.com/Satellite-im/Warp/issues/386 for details.

**Comment by dariusc93** [comment]
Noted in an earlier discussion, `ConversationDocument` has a field called restricted, which purpose is to store the group owner block list (and possibly any public key the group owner does not wish to allow to join the group).

https://github.com/Satellite-im/Warp/blob/9fe5488d1162c12a6b3866017076a81fe8812b8a/extensions/warp-ipfs/src/store/conversation.rs#L52

We can check this list when attempting to add a user to this list, possibly by adding a check to the following https://github.com/Satellite-im/Warp/blob/9fe5488d1162c12a6b3866017076a81fe8812b8a/extensions/warp-ipfs/src/store/message.rs#L2443-L2445

and for a safe measure, we can check this list again when receiving the event here

https://github.com/Satellite-im/Warp/blob/9fe5488d1162c12a6b3866017076a81fe8812b8a/extensions/warp-ipfs/src/store/message.rs#L1234-L1257

Note:
The group owner will emit an event if they block or unblock a user which would update the list in the conversation.

**Comment by zeenix** [comment]
@dariusc93 Thanks for the direct pointers. Very helpful and I believe the first part I've already done (since it was only 2-3 LOC in the end).

> The group owner will emit an event if they block or unblock a user which would update the list in the conversation.

From what I can tell from the code, this is already being done. However, I guess we also need to kick people from the group if the owner blocks them?

**Comment by dariusc93** [comment]
It needs to be checked to make sure that we dont add anybody from the restricted list.

**Comment by zeenix** [comment]
> It needs to be checked to make sure that we dont add anybody from the restricted list.

Ok but then what happens when owner blocks someone who is already a member? It would be a bit inconsistent if there is no consequence of that and we only honor owner's block list on addition.

**Comment by dariusc93** [comment]
> Ok but then what happens when owner blocks someone who is already a member? It would be a bit inconsistent if there is no consequence of that and we only honor owner's block list on addition.

Internally it already monitors for multipass events that are emitted when one is blocked so when the creator of the conversation blocks somebody, it would automatically remove them from the conversation and add them to that list. 

https://github.com/Satellite-im/Warp/blob/fbad79983fc69c76712c7ae2f0e22247feea069e/extensions/warp-ipfs/src/store/message.rs#L244-L272

**Comment by zeenix** [comment]
> Internally it already monitors for multipass events that are emitted when one is blocked so when the creator of the conversation blocks somebody, it would automatically remove them from the conversation and add them to that list.

Cool.
